src/control/server/storage/bdev/backend_json.go

```diff
@@ -217,7 +217,9 @@ func getSpdkConfigMethods(req *storage.BdevWriteConfigRequest) (sscs []*SpdkSubs
 		}
 
 		for index, dev := range tier.DeviceList.Devices() {
-			name := fmt.Sprintf("%s_%d_%d", req.Hostname, index, tier.Tier)
+			// Encode bdev tier info in RPC name field.
+			name := fmt.Sprintf("%s_%d_%d_%s", req.Hostname, index, tier.Tier,
+				tier.DeviceRoles.String())
```

Comment:
Not crazy about encoding even more information this way. Feels to me like these parameters want to be passed as a struct.

Reply:
there is some discussion over the decision to do this in the JIRA ticket. I won't repeat the rationale here but it's basically about duplication. Given the high priority off getting this communication in place I would prefer to keep this for the moment and differ moving the information elsewhere to a later stage. The consensus to carry the information this way was agreed on the ticket.

Reply:
OK. Fair enough.